OptimizelySDK/OptlyConfig/OptimizelyConfigService.cs

```diff
@@ -0,0 +1,226 @@
+        private IDictionary<string, OptimizelyExperiment> GetExperimentsMap(ProjectConfig projectConfig)
+        {
+            var experimentsMap = new Dictionary<string, OptimizelyExperiment>();
+            var rolloutExperimentIds = GetRolloutExperimentIds(projectConfig.Rollouts);
+            var featureVariableIdMap = GetVariableIdMap(projectConfig);
+
+            foreach (Experiment experiment in projectConfig.Experiments)
+            {
+                if (!rolloutExperimentIds.Contains(experiment.Id))
+                {
+                    var variationsMap = new Dictionary<string, OptimizelyVariation>();
+                    foreach (Variation variation in experiment.Variations)
+                    {
+                        var variablesMap = MergeFeatureVariables(projectConfig,
+                            featureVariableIdMap,
+                            experiment.Id,
+                            variation.FeatureEnabled ?? false,
+                            variation.FeatureVariableUsageInstances);
+
+                        var optimizelyVariation = new OptimizelyVariation(variation.Id,
+                            variation.Key,
+                            variation.FeatureEnabled,
+                            variablesMap);
+
+                        variationsMap.Add(variation.Key, optimizelyVariation);
+                    }
+                    var optimizelyExperiment = new OptimizelyExperiment(experiment.Id,
+                        experiment.Key,
+                        variationsMap);
+
+                    experimentsMap.Add(experiment.Key, optimizelyExperiment);
+                }
+            }
+
+            return experimentsMap;
+        }
+
+        /// <summary>
+        /// Make map of featureVariable which are associated with given feature experiment 
+        /// </summary>
+        /// <param name="projectConfig">The project config</param>
+        /// <param name="variableIdMap">Map containing variable ID as key and Object of featureVariable</param>
+        /// <param name="experimentId">experimentId of featureExperiment</param>
+        /// <param name="featureEnabled">featureEnabled of variation</param>
+        /// <param name="featureVariableUsageInstances">list of FeatureVariableUsage containing key and value</param>
+        /// <returns>Dictionary | Dictionary of FeatureVariable key and value as FeatureVariable object</returns>
+        private IDictionary<string, OptimizelyVariable> MergeFeatureVariables(
+           ProjectConfig projectConfig,
+           IDictionary<string, FeatureVariable> variableIdMap,
+           string experimentId,
+           bool featureEnabled,
+           List<FeatureVariableUsage> featureVariableUsageInstances)
+        {
+            var variablesMap = new Dictionary<string, OptimizelyVariable>();
+            var featureList = projectConfig.GetExperimentFeatureList(experimentId);
```

Comment:
`var featureList = projectConfig.GetExperimentFeatureList(experimentId) ?? null`

Reply:
there is no need of checking ?? null over here, in getExperimentFeatureList function I already added check for null